src/main/java/org/takes/facets/hamcrest/AbstractHmTextBody.java

```diff
@@ -74,18 +73,25 @@ public AbstractHmTextBody(final Matcher<String> body,
 
     @Override
     public final void describeTo(final Description description) {
-        description.appendDescriptionOf(this.body);
+        description.appendText("body: ").appendDescriptionOf(this.body);
     }
 
     @Override
     protected final boolean matchesSafely(final T item) {
         try {
-            return this.body.matches(this.text(item));
+            this.itext = this.text(item);
+            return this.body.matches(this.itext);
         } catch (final IOException ex) {
             throw new IllegalStateException(ex);
         }
     }
 
+    @Override
+    protected final void describeMismatchSafely(final T item, final
+        Description description) {
+        description.appendText("body was: ").appendText(this.itext);
```

Comment:
@olenagerasimova it should be `appendText(this.text(item))`, `this.itext` may point to wrong value, if `text(item) != this.itext`